### PR TITLE
Format code with ESLint (#5, partially #6)

### LINKED ISSUE #5: Format code in /static for consistency
Suggestions, please comment know if you disagree
- Use spaces after parameters
- Use 2xSpace for indentation instead of tabs
- Use single quote instead of double quote
- Use string interpolation instead of concatenation

I'd also suggest using ESLint to highlight deviations from the coding standard

### PR DESCRIPTION
**Pull request by nitzel**
- Added ESLint
- Added curly braces after `if/else/for` if they were missing
- Consistently using `tab`-indentation now
- Added `spaces` between parameters, operators etc
- Replaced `var` with `let` and `const`
- Turned all `double quotes` into `single quotes`
- Used `string interpolation` instead of concatenation
- Removed superfluous escaping backslashes from regexes
- Added globally used variables to `.eslintrc.globals` (there are a lot!)
- Added semicolons
- Opening and closing parenthesis for a function's parameters are now on a line each or on the same line

This closes #5 and makes some progress on #6

### CODE REVIEW

File: Static/js/chat.js [hunk 1]
```diff
@@ -1,265 +1,271 @@
...
+			$('.chattime').each(function (index) {
+				$(this).removeClass('hidden');
+			});
+		}
+	},
+	showchat: function () {
+		$('#chat-toggle-button').css('right', this.chat_width + 5);
+		$('#chat-toggle-text').html('>><br>c<br>h<br>a<br>t');
+		$('#chat').removeClass('hidden');
+		if (fixedcamera || true) {
```
Comment (nitzel) [inline Static/js/chat.js]: @NoHatCoder Do you know why this check exists here and in line 216? `|| true` looks like someone wanted to temporarily disable the condition - but if that's a permanent thing, we could remove the entire `if` statement.

File: Static/js/board.js [hunk 2]
```diff
@@ -1789,40 +1931,37 @@ var board = {
...
-		var playerTurn;
-		var moveNumber;
+		let playerTurn;
+		let moveNumber;
 
 		// simple RegEx for a basic TPS notation,
 		// but doesn't specify the details of the actual layout
-		var tpsRE = /\[(TPS\s*)?\"?\s*([,x12345678SC\/]+)(\s+([\d+]))?(\s+(\d+|-))?\s*\"?\s*\]/;
-		var result = tpsRE.exec(tps);
+		const tpsRE = /\[(TPS\s*)?"?\s*([,x12345678SC/]+)(\s+([\d+]))?(\s+(\d+|-))?\s*"?\s*\]/;
```
Comment (NoHatCoder) [inline Static/js/board.js]: Could we maybe not get rid of "unnecessary" escapes? Seeing / unescaped in the middle of an expression gives me the creeps.

File: Static/js/chat.js (hunk 1 above)
Comment (NoHatCoder) [reply Static/js/chat.js]: I wrote this while I was still unsure how all the cameras would work, I think we can just remove the if now.

**Comment by nitzel** [comment]
- [x] Remove all occurences of `// eslint-disable-next-line no-use-before-define` because it's disabled globally
- [x] Check if any other globally disabled rules still have related superfluous `// eslint-disable` comments
- [x] Remove `if(fixedcamera)` checks around `generateCamera()`
- [x] Escape `/` and `"` in regexes
- [x] ⛔  Enforce opening and closing parenthesis of `if` to be on their own lines or on the same line (like functions) 
  - I've had a look around [brace-style](https://eslint.org/docs/rules/brace-style), [operator-linebreak](https://eslint.org/docs/rules/operator-linebreak) and a few other but there's nothing like https://eslint.org/docs/rules/function-paren-newline for if instead of functions signatures.

**Pull request merged**